app/controllers/form_controller.rb

```diff
@@ -132,6 +158,14 @@ def successful_redirect_path
         return send("#{@log.model_name.param_key}_#{form.subsection_for_page(@page).id}_check_answers_path", @log)
       end
     end
+    if previous_interruption_screen_page_id.present?
+      if previous_interruption_screen_referrer.present?
+        return send("#{@log.class.name.underscore}_#{previous_interruption_screen_page_id}_path", @log, referrer: previous_interruption_screen_referrer)
+      else
+        return send("#{@log.class.name.underscore}_#{previous_interruption_screen_page_id}_path", @log)
+      end
```

Comment:
As this is the same route, you can compact nil values out:

```suggestion
     send("#{@log.class.name.underscore}_#{previous_interruption_screen_page_id}_path", @log, {referrer: previous_interruption_screen_referrer}.compact)
```

Reply:
Smart